db/schema.rb

```diff
@@ -24,20 +24,6 @@
     t.index ["record_type", "record_id", "name"], name: "index_action_text_rich_texts_uniqueness", unique: true
   end
 
-  create_table "active_admin_comments", force: :cascade do |t|
```

Comment:
not sure where this has gone so will fix

Reply:
I'm not sure why this table seems to drop in the migration - I can't see that I've mis-run anything. I'm not sure what this table is and it doesn't seem to exist in our Blazer schema. 

Should I keep the schema file like this or should I unstage this change (the table getting removed)?

Reply:
Ok good spot. So as context: Our database schema is _defined_ by migrations and _described_ by the schema. In practice, that means that every time you run `rails db:migrate`, the migrations get applied to your local database, and then Rails looks at the state of your database afterwards and describes what it sees in the schema file.

If you look through the migrations, you'll see we don't have one that creates an `active_admin_comments` table. So according to the definition of our database schema, this table doesn't exist. Every time you run migrations locally, you end up with a local database that doesn't have this table, so the schema file is updated without that table. And the schema file as it stands, with this table removed, is "correct".

So, how did this get in in the first place? Well, ActiveAdmin is an alternative to RailsAdmin that I played around with when first building the admin CMS. My guess is that I set up ActiveAdmin locally and ran migrations to set up a table that it used, then when I realised it didn't work I deleted the ActiveAdmin migrations without reverting them, and didn't notice that I still had this table defined in my local database, meaning that every time I ran any migration, the table would appear in the schema file (because it's next to a bunch of other auto-generated tables like "active_storage_attachments" etc). My bad, but because the schema file is largely for informational purposes only, it didn't have any knock-on consequences, which is why it wasn't noticed until someone _else_ started making PRs involving schema changes.

Reply:
to chat about when pairing